pyatlan/client/asset.py

```diff
@@ -850,40 +880,33 @@ def update_certificate(
         asset.certificate_status = certificate_status
         asset.name = name
         asset.certificate_status_message = message
+        if isinstance(asset, (AtlasGlossaryTerm, AtlasGlossaryCategory)):
+            self._update_glossary_anchor(asset, glossary_guid, asset_type.__name__)
         return self._update_asset_by_attribute(asset, asset_type, qualified_name)
 
-    def _update_asset_by_attribute(self, asset, asset_type, qualified_name: str):
-        query_params = {"attr:qualifiedName": qualified_name}
-        raw_json = self._client._call_api(
-            PARTIAL_UPDATE_ENTITY_BY_ATTRIBUTE.format_path_with_params(
-                asset_type.__name__
-            ),
-            query_params,
-            AssetRequest[Asset](entity=asset),
-        )
-        response = AssetMutationResponse(**raw_json)
-        if assets := response.assets_partially_updated(asset_type=asset_type):
-            return assets[0]
-        if assets := response.assets_updated(asset_type=asset_type):
-            return assets[0]
-        return None
-
     @validate_arguments()
     def remove_certificate(
```

Comment:
Please use the @overload annotation. See [PEP 3124](https://peps.python.org/pep-3124/) for more info. You will need to create another method with the original signature. workflow.py also uses this annotation.

Reply:
Hi @ErnestoLoma,

Thanks for the review. I've added the `@overload` decorator, following the example in `workflow.py`. Just a note, similar to `workflow.py`, I've kept type checks [within the method implementation](https://github.com/atlanhq/atlan-python/blob/8b1a2d0cc933e62a63ba009322255059f075c7ea/pyatlan/client/workflow.py#L121-L138). Could you please confirm that my new changes look good?